sdk/core/azure-core-management/src/main/java/com/azure/core/management/implementation/polling/PollerFactory.java

```diff
@@ -0,0 +1,269 @@
+// Licensed under the MIT License.
+
+package com.azure.core.management.implementation.polling;
+
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.polling.PollResult;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.polling.LongRunningOperationStatus;
+import com.azure.core.util.polling.PollResponse;
+import com.azure.core.util.polling.PollerFlux;
+import com.azure.core.util.polling.PollingContext;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * Factory to create PollerFlux for Azure resource manager (ARM) long-running-operation (LRO).
+ */
+public final class PollerFactory {
+    private static final ClientLogger logger = new ClientLogger(PollerFactory.class);
+    private static LongRunningOperationStatus LRO_CANCELLED = LongRunningOperationStatus.fromString("Cancelled",
+        true);
+
+    /**
+     * Creates a PollerFlux.
+     *
+     * @param serializerAdapter the serializer for any encoding and decoding
+     * @param pipeline the HttpPipeline for making any Http request (e.g. poll)
+     * @param pollResultType the type of the poll result, if no result is expecting then this should be Void.class
+     * @param finalResultType the type of the final result, if no result is expecting then this should be Void.class
+     * @param defaultPollInterval the default poll interval to use if service does not return retry-after
+     * @param lroInitOperation the function upon invoking should initiate the long-running-operation
+     * @param <T> the type of poll result
+     * @param <U> the type of final result
+     * @return PollerFlux
+     */
+    public static <T, U> PollerFlux<PollResult<T>, U> create(
+        SerializerAdapter serializerAdapter,
+        HttpPipeline pipeline,
+        Type pollResultType,
+        Type finalResultType,
+        Duration defaultPollInterval,
+        Function<PollingContext<PollResult<T>>, Mono<PollResult<T>>> lroInitOperation) {
+        Objects.requireNonNull(serializerAdapter, "'serializerAdapter' cannot be null.");
+        Objects.requireNonNull(pipeline, "'pipeline' cannot be null.");
+        Objects.requireNonNull(pollResultType, "'pollResultType' cannot be null.");
+        Objects.requireNonNull(finalResultType, "'finalResultType' cannot be null.");
+        Objects.requireNonNull(defaultPollInterval, "'defaultPollInterval' cannot be null.");
+        Objects.requireNonNull(lroInitOperation, "'lroInitOperation' cannot be null.");
+
+        return new PollerFlux<>(defaultPollInterval,
```

Comment:
The most struggle me when I tried this PR is about lroInitOperation that will succeed without polling operation.  

1. Current the PollerFlux/PollerFlux doesn't provide the short circuit when the init operation doesn't need polling operation. It will run the poll function anyway with delayed retry time.
2. Secondly, the Response<T> object from RestProxy will lost the string of the body, in some case, the some information will lost during pollingstate serialization and deserialization because of the properties are marked with WRITE_ONLY.

Reply:
1. Agree, we're missing a short circuit for synchronously completed LRO and imposes delay for the first poll that won't make any network call. Two options I could thinking of addressing with without breaking public surface: (1). Not to apply delay before the first poll and immediately call into poll function, it can check for completion state and return (2). Use PollerContext to signal synchronous completion, we check for that flag before starting polling loop.